app/precompiled.py

```diff
@@ -709,12 +709,23 @@ def _extract_text_from_page(page, rect):
     """
     words = page.get_text_words()
     mywords = [w for w in words if fitz.Rect(w[:4]).intersects(rect)]
+
+    def _get_address_from_get_textwords():
+        return page.get_text(clip=rect).strip()
+
     mywords.sort(key=itemgetter(-3, -2, -1))
     group = groupby(mywords, key=itemgetter(3))
     extracted_text = []
-    for _y1, gwords in group:
+    for _y2, gwords in group:
         extracted_text.append(" ".join(w[4] for w in gwords))
-    return "\n".join(extracted_text)
+    extracted_text = "\n".join(extracted_text)
+
+    if _get_address_from_get_textwords() != extracted_text:
+        # grouping by paragraph ended up different to grouping by y2. lets just log for now. we might want to swap over
+        # in the future but without knowing how much it changes we cant be sure
+        current_app.logger.info("Address extraction different between y2 and get_text")
```

Comment:
will we be able to tell for which letter it was, to see what the difference was? I vaguely remember we did some work to connect the whole request better in the logs?

Reply:
it's possible, but a little clunky. you can see the request ID on the logs in template-preview, and work backwards from there

[admin sanitise flow](https://kibana.logit.io/s/9423a789-282c-4113-908d-0be3b1bc9d1d/goto/740770eda421d4a3823ecd6512093e45)
[api async flow](https://kibana.logit.io/s/9423a789-282c-4113-908d-0be3b1bc9d1d/goto/7ab2dc601886ccd32cc8fdec11aeda1d)

both of these will lead back to a notification ID or an upload ID which you can use to find the actual file and see what's up